File: Annadata/app/src/main/java/com/mananhirak/annadata/foodcycle/FRAGMENT_ADAPTER.java
package com.mananhirak.annadata.foodcycle;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.mananhirak.annadata.fragments.BuyMarketFragment;
import com.mananhirak.annadata.fragments.PersonalFragment;

public class FRAGMENT_ADAPTER extends FragmentStateAdapter {
    public FRAGMENT_ADAPTER(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    public String TiffinLinkSellerId;

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 1) return new PersonalFragment();
        BuyMarketFragment buyMarketFragment = new BuyMarketFragment();
        if (TiffinLinkSellerId!=null) {
            Bundle bundle = new Bundle();
            bundle.putString("TiffinLinkSellerId", TiffinLinkSellerId);
            buyMarketFragment.setArguments(bundle);
        }
            return buyMarketFragment;
    }
    @Override
    public int getItemCount() {
        return 2;
    }
}

File: Annadata/app/src/main/java/com/manan/annadata/foodcycle/FRAGMENT_ADAPTER.kt
package com.manan.annadata.foodcycle

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.manan.annadata.fragments.BuyMarketFragment
import com.manan.annadata.fragments.PersonalFragment

class FRAGMENT_ADAPTER constructor(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    var TiffinLinkSellerId: String? = null
    public override fun createFragment(position: Int): Fragment {
        if (position == 1) return PersonalFragment()
        val buyMarketFragment: BuyMarketFragment = BuyMarketFragment()
        if (TiffinLinkSellerId != null) {
            val bundle: Bundle = Bundle()
            bundle.putString("TiffinLinkSellerId", TiffinLinkSellerId)
            buyMarketFragment.setArguments(bundle)
        }
        return buyMarketFragment
    }

    public override fun getItemCount(): Int {
        return 2
    }
}